salesagility/SuiteCRM
# Hotfix 7.8.x - Case Management AOP Improvements & Workflow Email Decimal Field Fix.

Fixes to bugs found in the AOP Case Management System and also emails sent from workflows dont populate the decimal field type.
## Description



##Workflow Email Field Type:##
When email template replace variable functionality, there was no option to replace decimal field type. There is currecny, dropdown, multi dropdown, int, image available but decimal type does not. I have added decimal field type. Now It is working.

##AOP Case Management Fixes:##
1) Created a new case from an inbound email but It did not update AOP_case_update entry and It did not send an email to the Assigned user

2) Even if there is no email template selected in AOP "User Email Template:" and "Contact Email Template:", the contact will receive a blank email.

## Motivation and Context

##Workflow Email Field Type:##
When sending emails from a workflow decimal field types are not populated.

##AOP Case Management Fixes:##
1. Inbound emails that generate cases need to update the case thread.
2. If there is no email template selected in the User Email Template field and Contact Email Template field an email should not be sent to the contact.

## How To Test This

##Workflow Email Field Type:##
1. Create a workflow to send an email template
2. Ensure the template contains a variable withe a filed type of decimal.
3. Ensure the test record has a value in this field.
4. Complete the workflow condition, the received email will not have the decimal field populated.

## Types of changes

- [X ] Bug fix (non-breaking change which fixes an issue)

### Final checklist


- [X ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [X ] My change requires a change to the documentation.
- [X ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.
